src/observable/subscribe.js

```diff
@@ -34,13 +34,23 @@ SubscribeObserver.prototype.event = function (t, x) {
 SubscribeObserver.prototype.end = function (t, x) {
   if (!this.disposable.disposed) {
     var s = this.subscriber
-    doDispose(this.fatalError, s, s.complete, s.error, this.disposable, x)
+    var fatalError = this.fatalError
+    Promise.resolve(this.disposable.dispose()).then(function () {
+      if (typeof s.complete === 'function') {
+        s.complete(x)
+      }
+    }).catch(function (e) {
+      throwError(e, s.error, fatalError)
```

Comment:
Ah, yes, I see. Thanks for clarifying.

In that case, I think I'd still prefer passing the subscriber as the 2nd arg rather than using `bind` in 2 places.